### ISSUE TITLE
Some modification required for source-dependent MC production

### ISSUE THREAD

**Issue by chaimain** [cta-observatory/lstmcpipe#191]
Following the discussion in #188 we will need to add `--source-dep` option for generating IRFs, and maybe some more things, as the discussion with @SeiyaNozaki progresses, regarding producing source dependent MC production.

**Comment by LukasNickel** [comment]
Choosing pointlike gammas for RF right?
Anything else?
Lets discuss it in the call

**Comment by SeiyaNozaki** [comment]
I think we need to modify for source-dependent analysis:
https://github.com/cta-observatory/lstmcpipe/blob/master/lstmcpipe/stages/mc_train.py#L52-L53
to use point gamma MC as RF training input
and
https://github.com/cta-observatory/lstmcpipe/blob/master/lstmcpipe/stages/mc_dl2_to_irfs.py#L284
to add `--source-dep` flag if soure-dep analysis is selected

**Comment by LukasNickel** [comment]
Pointlike gammas for all random forests. Is that a good idea?

Is it enough to put source dependent in the config and use the normal for dl1/dl2 or is there something missing?

**Comment by SeiyaNozaki** [comment]
> Pointlike gammas for all random forests. Is that a good idea?

yes, this is the way for source-dependent analysis. It would make some bias for proton/electrom MC, but we can obtain better performance for point gamma source.

> Is it enough to put source dependent in the config and use the normal for dl1/dl2 or is there something missing?

It should be enough only to activate source-dep flag in the config.

**Comment by vuillaut** [comment]
I believe this can now be easily supported by the new PathConfig logic.
Is there still a need @SeiyaNozaki @chaimain ?

**Comment by vuillaut** [comment]
Hi @SeiyaNozaki 
With the AllSky production, what gamma do you use for training for the source-dep analysis?

**Comment by SeiyaNozaki** [comment]
For all sky MC, I'm using diffuse gamma as RF training for source-dependent analysis (though I used point gamma before all sky MC).

**Comment by vuillaut** [comment]
> For all sky MC, I'm using diffuse gamma as RF training for source-dependent analysis (though I used point gamma before all sky MC).

ha ok great, so #375 is in line with your processing and it also means there is no changes required in lstmcpipe to process source-dependent !!
You can request a prod for your source-dep analysis, the only difference is in the lstchain config.
